pkg/manager/member/template.go

```diff
@@ -49,11 +49,24 @@ then
 fi
 
 # Use HOSTNAME if POD_NAME is unset for backward compatibility.
-POD_NAME=${POD_NAME:-$HOSTNAME}
+POD_NAME=${POD_NAME:-$HOSTNAME}{{ if .FormatClusterDomain }}
+pd_url="{{ .Path }}"
+encoded_domain_url=` + "`" + `echo $pd_url | base64 | tr "\n" " " | sed "s/ //g"` + "`" + `
+discovery_url="${CLUSTER_NAME}-discovery.${NAMESPACE}.svc{{ .FormatClusterDomain }}:10261"
+until result=$(wget -qO- -T 3 http://${discovery_url}/verify/${encoded_domain_url} 2>/dev/null | sed 's/http:\/\///g'); do
+echo "waiting for the verification of PD endpoints ..."
+sleep $((RANDOM % 5))
+done
+
+ARGS="--store=tikv \
+--advertise-address=${POD_NAME}.${HEADLESS_SERVICE_NAME}.${NAMESPACE}.svc{{ .FormatClusterDomain }} \
+--host=0.0.0.0 \
+--path=${result} \
+{{ else }}
```

Comment:
why only use the new `verify` service `if .FormatClusterDomain`?

Reply:
To avoid break back-compatibility. There is no need to verify the PD endpoint if there are no PeerMembers because there are no alternatives